sympy/matrices/expressions/matpow.py

```diff
@@ -15,8 +14,6 @@ def __new__(cls, base, exp, evaluate=False, **options):
         base = _sympify(base)
         if not base.is_Matrix:
             raise TypeError("MatPow base should be a matrix")
-        if not base.is_square:
-            raise NonSquareMatrixError("Power of non-square matrix %s" % base)
```

Comment:
Maybe `is_square` should use ternary logic and then this could be `if base.is_square is False`.